software/contrib/master_clock.py

```diff
@@ -0,0 +1,353 @@
+                # Only adjust values if k2 has moved. This avoids a potentially annoying UX
+                # self.activeOption != 3 / output 1 is disabled from configuration
+                if self.previousSelectedDivision != selectedDivision and self.activeOption != 3:
+                    self.outputDivisions[self.activeOption - 3] = selectedDivision
+                
+                self.previousSelectedDivision = selectedDivision
+            
+            self.previousActiveOption = self.activeOption
+                    
+        else:
+            configMarker = '.'
+        
+        oled.text(str(self.bpm) + ' bpm', 6, 0, 1)
+        oled.text(str(self.pulseWidthPercent) + ':' + str(str(self.pulseWidthMs)), 75, 0, 1)
+        oled.text('/' + str(self.outputDivisions[0]), 6, 12, 1)
+        oled.text('/' + str(self.outputDivisions[1]), 45, 12, 1)
+        oled.text('/' + str(self.outputDivisions[2]), 85, 12, 1)
+        oled.text('/' + str(self.outputDivisions[3]), 6, 24, 1)
+        oled.text('/' + str(self.outputDivisions[4]), 45, 24, 1)
+        oled.text('/' + str(self.outputDivisions[5]), 85, 24, 1)
+        oled.text(configMarker, self.markerPositions[self.activeOption-1][0], self.markerPositions[self.activeOption-1][1], 1)
+        oled.show() 
+
+    ''' Holds given output (cv) high for pulseWidthMs duration '''
+    async def outputPulse(self, cv):
+        cv.voltage(5)
+        await asyncio.sleep_ms(self.pulseWidthMs)
+        cv.off()
+
+    ''' Given a desired BPM, calculate the time to sleep between clock pulses '''
+    def calcSleepTime(self):
+        self.timeToSleepMs = int((60000 / self.bpm / self.CLOCKS_PER_QUARTER_NOTE))
+    
+    def checkForAinBPM(self):
+        val = 100 * ain.percent()
+        # If there is an analogue input voltage use that for BPM. clamp ensures it is higher than MIN and lower than MAX
+        if val > self.MIN_AIN_VOLTAGE:
+            self.bpm = clamp(int((((self.MAX_BPM) / 100) * val) + self.MIN_BPM), self.MIN_BPM, self.MAX_BPM)
+            self.calcSleepTime()
+            self.getPulseWidth()
+        else:
+            # No analog input, revert to last saved state
+            self.bpm = self.state.get("bpm", 100)
+            self.calcSleepTime()
+            self.getPulseWidth()
+
+    def getPulseWidth(self):
+        # Set max of self.MAX_PW_PERCENTAGE percent of total cycle time
+        self.MAX_PULSE_WIDTH = int(self.timeToSleepMs * self.MAX_PW_PERCENTAGE // 100)
+        # Calc pulse width in milliseconds given the desired percentage. clamp ensures it is higher than MIN and lower than MAX
+        self.pulseWidthMs = clamp((self.timeToSleepMs * (self.pulseWidthPercent)//100), self.MIN_PULSE_WIDTH, self.MAX_PULSE_WIDTH)
+
+    def computeGcd(self, x, y):
+        while(y):
+            x, y = y, x % y
+        return x
+
+    def lcm(self, li):
+        lcm = 1
+        for item in li:
+            lcm = lcm*item//self.computeGcd(lcm, item)
```

Comment:
```suggestion
            lcm = lcm*item//max(self.computeGcd(lcm, item), 1)
```

This would prevent the risk of a divide by zero in this function regardless of what is returned from computeGcd